Tests/FluentAssertions.Specs/Collections/CollectionAssertionSpecs.AllSatisfy.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions.Execution;
+using FluentAssertions.Specs.Equivalency;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Collections
+{
+    /// <content>
+    /// The AllSatisfy specs.
+    /// </content>
+    public partial class CollectionAssertionSpecs
+    {
+        #region All Satisfy
+
+        [Fact]
+        public void When_collection_asserting_against_null_inspector_it_should_throw_with_clear_explanation()
```

Comment:
🔧 Given that the scope of this class is clearly about the `AllSatisfy` method for generic collections, you can make the names much more concise. E.g. `A_null_inspector_is_not_valid`